roles/openshift_monitor_availability/files/web-console-smoke-test.yaml

```diff
@@ -0,0 +1,117 @@
+    openshift.io/support-url: https://access.redhat.com
+    openshift.io/provider-display-name: Red Hat, Inc.
+    description: >
+      Runs smoke tests in a headless browser against the web console at a provided URL.
+
+      For more information about using this template, see
+      https://github.com/benjaminapetersen/origin-web-console-smoke-test/blob/master/README.md
+objects:
+- apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: web-console-smoke-test
+    namespace: ${NAMESPACE}
+    labels:
+      app: web-console-smoke-test
+  spec:
+    # replica set
+    replicas: ${REPLICA_COUNT}
+    selector:
+      matchLabels:
+        app: web-console-smoke-test
+    template:
+      metadata:
+        labels:
+          app: web-console-smoke-test
+      spec:
+        containers:
+        - name: web-console-smoke-test
+          image: ${IMAGE}
+          imagePullPolicy: ${PULL_POLICY}
+          env:
+          - name: TOKEN
+            value: ${TOKEN}
+          - name: CONSOLE_URL
+            value: ${CONSOLE_URL}
+          - name: CONSOLE_USER
+            value: ${CONSOLE_USER}
+          - name: CONSOLE_PASSWORD
+            value: ${CONSOLE_PASSWORD}
+          - name: TEST_INTERVAL_MINUTES
+            value: ${TEST_INTERVAL_MINUTES}
+          ports:
+          - containerPort: 3000
+          restartPolicy: Always
+          volumeMounts:
+            - mountPath: /etc/tls-certs
+              name: web-console-smoke-test-serving-cert
+              readOnly: true
+        volumes:
+        - name: web-console-smoke-test-serving-cert
+          secret:
+            defaultMode: 400
+            secretName: web-console-smoke-test-serving-cert
+
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: web-console-smoke-test
+    namespace: ${NAMESPACE}
+    annotations:
+      service.alpha.openshift.io/serving-cert-secret-name: web-console-smoke-test-serving-cert
+      prometheus.io/scrape: "true"
+      prometheus.io/scheme: https
+  spec:
+    ports:
+    - name: web
+      protocol: TCP
+      port: 443
+      targetPort: 3000
+      nodePort: 0
+    selector:
+      app: web-console-smoke-test
+    type: ClusterIP
+    sessionAffinity: None
+  status:
+    loadBalancer: {}
+parameters:
+- name: IMAGE
+# TODO: update this URL when we migrate up the repo
+  value: benjaminapetersen/origin-web-console-smoke-test:latest
+- name: NAMESPACE
+  # This namespace cannot be changed. Only `openshift-web-console-smoke-test` is supported.
+  value: openshift-web-console-smoke-test
+- name: REPLICA_COUNT
+  value: "1"
+- name: PULL_POLICY
+  value: IfNotPresent
+- name: CONSOLE_URL
+  description: The location of the web console, in the format of https://<machine-ip>:8443
+  required: true
+- name: CONSOLE_USER
+  description: An optional username
+- name: CONSOLE_PASSWORD
+  description: An optional password
+- name: TOKEN
```

Comment:
I think CONSOLE_USER, CONSOLE_PASSWORD, and TOKEN need to be secrets.

Reply:
Fields of a single secret, rather.

Reply:
+1, these should be secrets

Reply:
Actually, is the TOKEN even needed as an env var if we use the service account token?

You probably want to create the service account and give it the basic access it needs for web console as part of this template.

Reply:
Probably not in the template.  Similar to `USER_*`, we've used a lot for dev purposes.

Reply:
@spadgett using the `service account token` will be fine if the smoke tests are deployed to test the *same cluster*, but would we ever want to test *across clusters*?  At that point, I don't believe the service account token will be useful.  I think we may have left this in as a reminder to explore that use case.

Reply:
The tests are intended to be cluster-local. Does that help narrow the scope?

Reply:
Great, that is definitely helpful.